src/server/wesnothd/game.hpp

```diff
@@ -117,199 +180,352 @@ class game
-	 *                            started game.
+	 * @param player The player to remove.
+	 * @param disconnect If the player disconnected from the server entirely.
+	 * @param destruct If the game is ending as well.
+	 * @return True if the player's removal ends the game. That is, if there are no more players or the host left on a not yet started game.
 	 */
 	bool remove_player(player_iterator player, const bool disconnect = false, const bool destruct = false);
 
-	/** Adds players and observers into one vector and returns that. */
+	/**
+	 * @return A vector containing all players and observers currently in this game.
+	 */
 	const user_vector all_game_users() const;
 
+	/**
+	 * Starts the game (if a new game) or starts the next scenario of an MP campaign.
+	 * @param starter The game's host.
+	 */
 	void start_game(player_iterator starter);
 
-	// this is performed just before starting and before [start_game] signal
-	// send scenario_diff's specific to each client so that they locally
-	// control their human sides
+	/**
+	 * This is performed just before starting and before the [start_game] signal
+	 * sends [scenario_diff]s specific to each client so that they locally control their human sides.
+	 */
 	void perform_controller_tweaks();
 
-	void update_game();
-
-	/** A user (player only?) asks for the next scenario to advance to. */
-	void load_next_scenario(player_iterator user); // const
-
-	// iceiceice: I unmarked this const because I want to send and record server messages when I fail to tweak sides
-	// properly
+	/**
+	 * A user (player only?) asks for the next scenario to advance to.
+	 *
+	 * @param user The user asking for the next scenario.
+	 */
+	void load_next_scenario(player_iterator user);
 
 	/** Resets the side configuration according to the scenario data. */
 	void update_side_data();
 
-	/** Let's a player owning a side give it to another player or observer. */
+	/**
+	 * Lets a player owning a side give it to another player or observer.
+	 *
+	 * @param player The player owning the side.
+	 * @param cfg The node containing the transfer information.
+	 */
 	void transfer_side_control(player_iterator player, const simple_wml::node& cfg);
 
-	void process_message(simple_wml::document& data, player_iterator);
+	/**
+	 * Sends an ingame message to all other players.
+	 *
+	 * @param data The message to send.
+	 * @param user The user sending the message.
+	 */
+	void process_message(simple_wml::document& data, player_iterator user);
 
 	/**
 	 * Handles [end_turn], repackages [commands] with private [speak]s in them
 	 * and sends the data.
 	 * Also filters commands from all but the current player.
 	 * Currently removes all commands but [speak] for observers and all but
-	 * [speak], [label] and [rename] for players.
+	 * [speak], [label], and [rename] for players.
 	 *
-	 * @returns                   True if the turn ended.
+	 * @param data The turn commands.
+	 * @param user The user whose turn is being processed.
```

Comment:
So something like:
`@param user The user who sent a command to be processed during the turn. This may not be the player whose turn it currently is.`
?

Reply:
Yes.